pe.c

```diff
@@ -1080,6 +1082,22 @@ handle_image (void *data, unsigned int datasize,
 			return EFI_UNSUPPORTED;
 		}
 
+		efi_status = parse_sbat_var(&sbat_var_entries);
+		/*
+		 * Until a SBAT variable is installed into the systems, it is expected that
+		 * attempting to parse the variable will fail with an EFI_NOT_FOUND error.
+		 *
+		 * Do not consider that error fatal for now.
+		 */
+		if (EFI_ERROR(efi_status) && efi_status != EFI_NOT_FOUND) {
+			perror(L"Parsing SBAT variable failed: %r\n",
+			       efi_status);
+			return efi_status;
+		}
+
+		if (efi_status == EFI_SUCCESS)
+			efi_status = verify_sbat(&sbat, &sbat_var_entries);
+
 		efi_status = verify_buffer(data, datasize,
```

Comment:
Isn't the status returned from verify_sbat() being ignored here? Shouldn't verify_buffer() be skipped if verify_sbat() fails?

Reply:
Actually we did had a verification here, this one: https://github.com/martinezjavier/shim/commit/06ee1d9f8b06150a6ea4c7893325fdfbf8b29182#diff-7de403009b80237157eccb017ada7717e5d12c3b24afe8783e7370363191c073R1873 but Javier removed it during his rebase, i guess he had something else in mind. Anyway, i am keeping an eye on this, since most likely we need to readd it.

Reply:
Right, that was a mistake during my rebase when resolving the conflicts. I meant to keep the check for `verify_sbat()` there.

Reply:
should be fixed, though please double check current code. Right now, we will fail validation if there is no SBAT header in binary, even if binary is enrolled by hash.